api/src/apps/routes/logs.rs

```diff
@@ -0,0 +1,374 @@
+        app_name,
+        service_name,
+        limit: log_query.limit,
+        as_attachment: log_query.as_attachment,
+    })
+}
+
+#[get(
+    "/<app_name>/logs/<service_name>?<log_query..>",
+    format = "text/event-stream",
+    rank = 2
+)]
+pub(super) async fn stream_logs<'r>(
+    app_name: Result<AppName, AppNameError>,
+    service_name: &'r str,
+    log_query: LogQuery,
+    apps: &'r State<Arc<Apps>>,
+) -> HttpResult<EventStream![Event + 'r]> {
+    let app_name = app_name.unwrap();
+    let since = match &log_query.since {
+        None => None,
+        Some(since) => match DateTime::parse_from_rfc3339(&since) {
+            Ok(since) => Some(since),
+            Err(err) => {
+                return Err(
+                    HttpApiProblem::with_title(http_api_problem::StatusCode::BAD_REQUEST)
+                        .detail(format!("{}", err))
+                        .into(),
+                );
+            }
+        },
+    };
+
+    Ok(EventStream! {
+        let mut log_chunk = apps
+            .stream_logs(&app_name, service_name, &since, &log_query.limit)
+            .await;
+
+        while let Some(result) = log_chunk.as_mut().next().await {
+            match result {
+                Ok((_, log_line)) => yield Event::data(log_line),
+                Err(_e) => {
+                    break;
+                }
+            }
+        }
+    })
+}
+
+pub struct LogsResponse<'a> {
+    log_chunk: Option<LogChunk>,
+    app_name: AppName,
+    service_name: &'a str,
+    limit: Option<usize>,
+    as_attachment: bool,
+}
+
+impl<'r, 'o: 'r> Responder<'r, 'o> for LogsResponse<'r> {
+    fn respond_to(self, _request: &'r Request) -> Result<Response<'o>, Status> {
+        use std::io::Cursor;
+        let log_chunk = match self.log_chunk {
+            None => {
+                let payload = HttpApiProblem::with_title(http_api_problem::StatusCode::NOT_FOUND)
+                    .json_bytes();
+                return Response::build()
+                    .status(Status::NotFound)
+                    .raw_header("Content-type", "application/problem+json")
+                    .sized_body(payload.len(), Cursor::new(payload))
+                    .ok();
+            }
+            Some(log_chunk) => log_chunk,
+        };
+
+        let from = *log_chunk.until() + chrono::Duration::milliseconds(1);
+
+        let next_logs_url = match self.limit {
+            Some(limit) => format!(
+                "/api/apps/{}/logs/{}?limit={}&since={}",
+                self.app_name,
+                self.service_name,
+                limit,
+                RawStr::new(&from.to_rfc3339()).percent_encode(),
+            ),
+            None => format!(
+                "/api/apps/{}/logs/{}?since={}",
+                self.app_name,
+                self.service_name,
+                RawStr::new(&from.to_rfc3339()).percent_encode(),
+            ),
+        };
+
+        let content_dispositon_value = if self.as_attachment {
```

Comment:
Self-Correction :) - My initial typo has carried through.
Please correct this - `content_disposition_value`